### ISSUE TITLE
🐞 Golang modules error while executing ./dagger version

### ISSUE THREAD

**Issue by slumbering** [dagger/dagger#6653]
### What is the issue?

The error occurs when I follow the section `export-the-build-directory` from the docs: https://docs.dagger.io/zenith/quickstart/853930/directories/#export-the-build-directory

I'm getting an error when trying to check the Dagger version:
```
$ ./dagger version
zsh: exec format error: ./dagger
```

cc @jedevc 

### Dagger version

dagger v0.9.9 (registry.dagger.io/engine) darwin/arm64

### Steps to reproduce

> [!IMPORTANT]
The issue seems to occur only on an Apple Silicon M1 device.

- Open a terminal
- Run this command first:
```
$ dagger -m github.com/kpenfound/dagger-modules/golang --proj https://github.com/dagger/dagger call build --args ./cmd/dagger export --path ./my-dagger-build
```
- Then run: 
```
$ cd my-dagger-build
$ ./dagger version
```

### Log output

zsh: exec format error: ./dagger

**Comment by jedevc** [comment]
Ooh, potentially this is part of the issue: https://github.com/kpenfound/dagger-modules/blob/e13cebb81cc0adb30ad9af533f8171a66b697102/golang/main.go#L39-L53

This should probably set `GOOS` as well as `GOARCH` @kpenfound? (similar to how `BuildRemote` works in the code below it).

I think we'd need to also update the docs to add `--arch` and `--platform`.

---

Aside: I *wonder* if we might want to expose the client's architecture to modules? Obviously, it's an information leak, and part of the "ambient system access problem", but:
1. this is super minimal information
2. modules could use this, if the aim is to export something to the client

Perhaps, we could have an `ExportTarget` or similar? Then when we know we're exporting files, we could send the client architecture that we're exporting to (though I wonder if this potentially makes ordering a bit tricky).

**Comment by kpenfound** [comment]
@jedevc yeah this definitely feels like it's a `GOOS` issue. I'll update the module!

**Comment by kpenfound** [comment]
I've added the optional `os` arg to the build command:
`dagger -m github.com/kpenfound/dagger-modules/golang --proj https://github.com/dagger/dagger call build --args ./cmd/dagger --os darwin export --path ./my-dagger-build`

**Comment by jedevc** [comment]
Thanks @kpenfound! Not quite sure how we should surface this in the docs page though now:
- Do we add a note about adding `--os`/`--arch` for cases where it needs to be included?
- Or do we just add it in, and use some shell magic to get the right value? (something like `uname -o`)

**Comment by kpenfound** [comment]
@jedevc in general it would be useful to query the host os/arch for a lot of reasons, since within the module sandbox it'll always be `linux`
